libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -762,20 +776,29 @@ def _generate_with_cache(
 
         # If stream is not explicitly set, check if implicitly requested by
         # astream_events() or astream_log(). Bail out if _stream not implemented
-        if type(self)._stream != BaseChatModel._stream and kwargs.pop(
-            "stream",
-            (
-                next(
-                    (
-                        True
-                        for h in run_manager.handlers
-                        if isinstance(h, _StreamingCallbackHandler)
-                    ),
-                    False,
-                )
-                if run_manager
-                else False
-            ),
+        if (
```

Comment:
could we factor this out into a method (with an async flag)

Reply:
ya